planet/clients/data.py

```diff
@@ -85,12 +85,14 @@ async def _do_request(self, request: Request) -> Response:
         """
         return await self._session.request(request)
 
-    async def quick_search(self,
-                           item_types: typing.List[str],
-                           search_filter: dict,
-                           name: str = None,
-                           sort: str = None,
-                           limit: int = None) -> typing.AsyncIterator[dict]:
+    async def quick_search(
+            self,
+            item_types: typing.List[str],
+            search_filter: dict,
+            name: str = None,
+            sort: str = None,
+            limit: typing.Union[int,
```

Comment:
I'm curious where the need for supporting `limit=None` comes from?

Reply:
so that even if limit is not specified (or set to None), the default of 100 items is still supported. We do a similar thing in `planet/clients/orders.py::list_orders` for the limit functionality there.

Reply:
of course!!